python/lsst/fgcmcal/utilities.py

```diff
@@ -811,7 +818,12 @@ def computeApertureRadiusFromDataRef(dataRef, fluxField):
        or associated slot.
     """
     # TODO: Move this method to more general stack method in DM-25775
-    datasetType = dataRef.butlerSubset.datasetType
+    try:
+        # Gen2 dataRef
+        datasetType = dataRef.butlerSubset.datasetType
+    except AttributeError:
+        # Gen3 dataRef
+        datasetType = dataRef.ref.datasetType.name
```

Comment:
You seem to be inconsistent in your files between this style check and isinstance checks.